frontend/src/pages/Account.vue

```diff
@@ -105,7 +276,109 @@ export default {
     },
     expiresAt () {
       return moment.duration(this.user.exp - Math.floor(Date.now() / 1000), 'seconds').humanize(true)
+    },
+    projectNames () {
+      const names = map(this.projectList, 'metadata.name').sort()
+      names.unshift('')
+      return names
+    },
+    kubeconfigFilename () {
+      if (this.projectName) {
+        return `kubeconfig-garden-${this.projectName}.yaml`
+      }
+      return 'kubeconfig-garden.yaml'
+    },
+    kubeconfig () {
+      const project = find(this.projectList, ['metadata.name', this.projectName])
+      const name = 'garden-' + get(project, 'metadata.name', 'none')
+      const namespace = get(project, 'metadata.namespace')
+      const {
+        server,
+        certificateAuthorityData,
+        insecureSkipTlsVerify,
+        oidc: {
+          issuerUrl,
+          clientId,
+          clientSecret,
+          extraScopes = []
+        } = {}
+      } = this.kubeconfigData || {}
+      const cluster = {
+        server
+      }
+      if (certificateAuthorityData) {
+        cluster['certificate-authority-data'] = certificateAuthorityData
+      } else if (insecureSkipTlsVerify) {
+        cluster['insecure-skip-tls-verify'] = true
+      }
+      const context = {
+        cluster: name,
+        user: 'oidc-login'
+      }
+      if (namespace) {
+        context.namespace = namespace
+      }
+      const args = [
+        'oidc-login',
+        'get-token',
+        '--oidc-issuer-url=' + issuerUrl,
+        '--oidc-client-id=' + clientId,
```

Comment:
Isn‘t the client-id optional? If it is not set, will it be `—oidc-client-id=undefined` ?